### PR TITLE
gh-99677: Deduplicate self-type in `mro` in `inspect._getmembers`

### PR DESCRIPTION
**Pull request by sobolevn**
I am not adding news, because there are no user-facing changes.

* Issue: gh-99677

### CODE REVIEW

**Comment by sobolevn** [comment]
@larryhastings looks like you are one of the most recent commiters to `inspect`. Maybe you would have time to take a look? :)

**Review by kumaraditya303 — APPROVED** [review]

File: Lib/inspect.py [hunk 1]
```diff
@@ -537,7 +537,7 @@ def _getmembers(object, predicate, getter):
     processed = set()
     names = dir(object)
     if isclass(object):
-        mro = (object,) + getmro(object)
```
Comment (kumaraditya303) [inline Lib/inspect.py:540]: MRO always contains the type so this is redundant.

Comment (iritkatriel) [reply Lib/inspect.py:540]: Is this not covered by any test?

Comment (kumaraditya303) [reply Lib/inspect.py:540]: This is covered in test. Adding object to the tuple of bases is redundant here and leads to object occurring twice in mro. `getmro` already returns `object` in mro.

**Review by iritkatriel — APPROVED** [review]

**Comment by sobolevn** [comment]
Thanks for the review! 👍

**Pull request merged**